graphs/src/anemoi/graphs/processors/post_process.py

```diff
@@ -131,32 +149,95 @@ class RemoveUnconnectedNodes(BaseNodeMaskingProcessor):
+        nodes = graph[nodes_name]
         connected_mask = torch.zeros(nodes.num_nodes, dtype=torch.bool)
 
         if self.ignore is not None:
             LOGGER.info(f"The nodes with {self.ignore}=True will not be removed.")
             connected_mask[nodes[self.ignore].bool().squeeze()] = True
 
         for (source_name, _, target_name), edges in graph.edge_items():
-            if source_name == self.nodes_name:
+            if source_name == nodes_name:
                 connected_mask[edges.edge_index[0]] = True
 
-            if target_name == self.nodes_name:
+            if target_name == nodes_name:
                 connected_mask[edges.edge_index[1]] = True
 
         return connected_mask
 
 
+class SubsetNodesInArea(BaseNodeMaskingProcessor):
+    """Remove unconnected nodes in the graph.
+
+    Attributes
+    ----------
+    nodes_name : str | Iterable[str]
+        Name of the unconnected nodes to remove.
+    area : tuple[float, float, float, float]
+        Area of interest to crop the nodes, (north, west, south, east).
+    save_mask_indices_to_attr : str, optional
+        Name of the attribute to save the mask indices. If provided,
+        the indices of the kept nodes will be saved in this attribute.
+
+    Methods
+    -------
+    compute_mask(graph)
+        Compute the mask of the connected nodes.
+    prune_graph(graph, mask)
+        Prune the nodes with the specified mask.
+    add_attribute(graph, mask)
+        Add an attribute of the mask indices as node attribute.
+    update_graph(graph)
+        Post-process the graph.
+    """
+
+    def __init__(
+        self,
+        nodes_name: Union[str, Iterable[str]],
+        area: tuple[float, float, float, float],
+        save_mask_indices_to_attr: str | None = None,
+    ) -> None:
+        super().__init__(nodes_name, save_mask_indices_to_attr)
+        self.area = area
+
+    @staticmethod
+    def points_inside_area(
+        lats: torch.Tensor, lons: torch.Tensor, area: tuple[float, float, float, float]
+    ) -> torch.Tensor:
+        north, west, south, east = area
+
+        # Convert area limits to [-90, 90]x[0,360]
+        west, east = west % 360, (east - 1) % 360 + 1
+        assert (west - east) % 360 != 0, "West and East limits must be different."
+
+        # Convert lat & lons to [-90, 90]x[0,360)
+        lons %= 360
+
+        in_lat_range = (lats >= south) & (lats <= north)
+        if west < east:
+            in_lon_range = (lons >= west) & (lons <= east)
+        else:
+            # e.g: W=-10,E=10. They will be mapped to west=350, east=10.
+            in_lon_range = (lons >= west) | (lons <= east)
+        return in_lat_range & in_lon_range
+
+    def compute_mask(self, graph: HeteroData, nodes_name: str) -> torch.Tensor:
+        """Compute the mask of connected nodes."""
+        latlons = graph[nodes_name].x * 180 / torch.pi
+        mask = SubsetNodesInArea.points_inside_area(lats=latlons[:, 0], lons=latlons[:, 1], area=self.area)
```

Comment:
Do you think we will be using this method somewhere else or what's the motivation for making it a staticmethod?